test/extended/builds/forcepull.go

```diff
@@ -1,183 +1,179 @@
-			g.It("\n FORCE PULL TEST dock true", func() {
-				fpTrueDock := exutil.FixturePath("fixtures", "forcepull-true-dock.json")
-				g.By(fmt.Sprintf("\n%s FORCE PULL TEST dock true:  calling create on %s", time.Now().Format(time.RFC850), fpTrueDock))
-				exutil.StartBuild(fpTrueDock, buildPrefix, oc)
+			doTest(fullNameDS, buildPrefixTS, buildNameTS, "s2i true app/lang build", false, oc)
 
-				exutil.WaitForBuild("FORCE PULL TEST dock true", buildName, oc)
+			g.By("when docker force pull is false")
 
-				exutil.VerifyImagesDifferent(s2iDockBldr, custBldr, "FORCE PULL TEST docker true:  ")
+			doTest(fullNameDS, buildPrefixFD, buildNameFD, "dock false app/lang build", true, oc)
 
-			})
-		})
-	})
-
-	g.Describe("\n FORCE PULL TEST:  Force pull and custom builder", func() {
-		// corrupt the custom builder image
-		g.BeforeEach(func() {
-			exutil.CorruptImage(custBldr, s2iDockBldr, "custom")
-		})
-
-		g.AfterEach(func() {
-			exutil.ResetImage(resetData)
-		})
+			g.By("docker when force pull is true")
 
-		g.Context("\n FORCE PULL TEST:  when custom force pull is false and the image is bad", func() {
-			g.It("\nFORCE PULL TEST cust false", func() {
-				fpFalseCust := exutil.FixturePath("fixtures", "forcepull-false-cust.json")
-				g.By(fmt.Sprintf("\n%s FORCE PULL TEST cust false:  calling create on %s", time.Now().Format(time.RFC850), fpFalseCust))
-				exutil.StartBuild(fpFalseCust, buildPrefix, oc)
+			doTest(fullNameDS, buildPrefixTD, buildNameTD, "dock true app/lang build", false, oc)
 
-				g.By("\nFORCE PULL TEST cust false:  expecting the image is not refreshed")
+			g.By("when custom force pull is false")
 
-				exutil.WaitForBuild("FORCE PULL TEST cust false", buildName, oc)
+			doTest(fullNameC, buildPrefixFC, buildNameFC, "cust false app/lang build", true, oc)
 
-				exutil.VerifyImagesSame(s2iDockBldr, custBldr, "FORCE PULL TEST custom false:  ")
-			})
-		})
-
-		g.Context("\n FORCE PULL TEST:  when custom force pull is true and the image is bad", func() {
-			g.It("\n FORCE PULL TEST cust true", func() {
-				fpTrueCust := exutil.FixturePath("fixtures", "forcepull-true-cust.json")
-				g.By(fmt.Sprintf("\n%s FORCE PULL TEST cust true:  calling create on %s", time.Now().Format(time.RFC850), fpTrueCust))
-				exutil.StartBuild(fpTrueCust, buildPrefix, oc)
-
-				g.By("\n FORCE PULL TEST cust true:  expecting the image is refreshed")
+			g.By("when custom force pull is true")
 
-				exutil.WaitForBuild("FORCE PULL TEST cust true", buildName, oc)
+			doTest(fullNameC, buildPrefixTC, buildNameTC, "cust true app/lang build", false, oc)
 
-				exutil.VerifyImagesDifferent(s2iDockBldr, custBldr, "FORCE PULL TEST custom true:  ")
-
-			})
+			// dump the image streams for debug compares
+			imageStreamOutput, err := oc.Run("get").Args("is", "-o", "json", "--config", exutil.KubeConfigPath()).Output()
+			g.By(fmt.Sprintf("image stream after test run err %v and output %s ", err, imageStreamOutput))
```

Comment:
can we get this output to display only if the test failed?  does gingko automatically buffer all fmt.Print output and then only display it if the test fails?  if so that'd be the way i'd prefer to do this.